thread/thread.cpp

```diff
@@ -922,7 +918,24 @@ R"(
                      stack_size, least_stack_size, least_stack_size);
             stack_size = least_stack_size;
         }
-        char* ptr = (char*)photon_thread_alloc(stack_size);
+        // char* ptr = (char*)photon_thread_alloc(stack_size);
+        struct pthread* pd;
+        char* ptr;  // ptr to stack
+        size_t pstacksize;
+        pthread_attr_t attr;
+        pthread_attr_init(&attr);
+        allocate_stack((struct pthread_attr*)&attr, &pd, (void**)&ptr, &pstacksize);
```

Comment:
We must make use of ```allocate_thread()``` in lib pthread to allocate the thread, TCB (that compatible to threading facilities) and TLS for all modules loaded at startup time.